Makefile

```diff
@@ -1,52 +1,386 @@
 else
+X ?=
+BUILD_MAIN ?= ./git-lfs.go
+endif
+
+# BUILD is a macro used to build a single binary of Git LFS using the above
+# LD_FLAGS and GC_FLAGS.
+#
+# It takes three arguments:
+#
+# 	$(1) - a valid GOOS value, or empty-string
+# 	$(2) - a valid GOARCH value, or empty-string
+# 	$(3) - an optional program extension. If $(3) is given as '-foo', then the
+# 	       program will be written to bin/git-lfs-foo.
+#
+# It uses BUILD_MAIN as defined above to specify the entrypoint for building Git
+# LFS.
+BUILD = GOOS=$(1) GOARCH=$(2) \
+	$(GO) build \
+	-ldflags="$(LD_FLAGS)" \
+	-gcflags="$(GC_FLAGS)" \
+	-o ./bin/git-lfs$(3) $(BUILD_MAIN)
+
+# BUILD_TARGETS is the set of all platforms and architectures that Git LFS is
+# built for.
+BUILD_TARGETS = bin/git-lfs-darwin-amd64 bin/git-lfs-darwin-386 \
+	bin/git-lfs-linux-amd64 bin/git-lfs-linux-386 \
+	bin/git-lfs-freebsd-amd64 bin/git-lfs-freebsd-386 \
+	bin/git-lfs-windows-amd64.exe bin/git-lfs-windows-386.exe
+
+# Targets 'all' and 'build' build binaries of Git LFS for the above release
+# matrix.
+.PHONY : all build
+all build : $(BUILD_TARGETS)
+
+# The following bin/git-lfs-% targets make a single binary compilation of Git
+# LFS for a specific operating system and architecture pair.
+#
+# They function by translating target names into arguments for the above BUILD
+# builtin, and appending the appropriate suffix to the build target.
+#
+# On Windows, they also depend on the resource.syso target, which installs and
+# embeds the versioninfo into the binary.
+bin/git-lfs-darwin-amd64 : $(SOURCES)
+	$(call BUILD,darwin,amd64,-darwin-amd64)
+bin/git-lfs-darwin-386 : $(SOURCES)
+	$(call BUILD,darwin,386,-darwin-386)
+bin/git-lfs-linux-amd64 : $(SOURCES)
+	$(call BUILD,linux,amd64,-linux-amd64)
+bin/git-lfs-linux-386 : $(SOURCES)
+	$(call BUILD,linux,386,-linux-386)
+bin/git-lfs-freebsd-amd64 : $(SOURCES)
+	$(call BUILD,freebsd,amd64,-freebsd-amd64)
+bin/git-lfs-freebsd-386 : $(SOURCES)
+	$(call BUILD,freebsd,386,-freebsd-386)
+bin/git-lfs-windows-amd64.exe : resource.syso $(SOURCES)
+	$(call BUILD,windows,amd64,-windows-amd64.exe)
+bin/git-lfs-windows-386.exe : resource.syso $(SOURCES)
+	$(call BUILD,windows,386,-windows-386.exe)
+
+# .DEFAULT_GOAL sets the operating system-appropriate Git LFS binary as the
+# default output of 'make'.
+.DEFAULT_GOAL := bin/git-lfs$(X)
+
+# bin/git-lfs targets the default output of Git LFS on non-Windows operating
+# systems, and respects the build knobs as above.
+bin/git-lfs : $(SOURCES) fmt
+	$(call BUILD,$(GOOS),$(GOARCH),)
 
-all : bin/git-lfs
+# bin/git-lfs targets the default output of Git LFS on Windows systems, and
+# respects the build knobs as above.
+bin/git-lfs.exe : $(SOURCES) resource.syso
+	$(call BUILD,$(GOOS),$(GOARCH),.exe)
 
-include $(MAKEFILE_GEN)
+# resource.syso install the 'goversioninfo' command and uses it in order to
+# generate a binary that has information included necessary to create the
+# Windows installer.
+resource.syso:
```

Comment:
🤔, hmm -- I'm not sure why this is happening! I cleared out `$GOPATH/bin`, and was able to build this successfully:

```ShellSession
$ make -B resource.syso
go get github.com/josephspurrier/goversioninfo/cmd/goversioninfo
go generate
$ echo $?
0
```

Do you want to debug this a little further offline together some later time? My hunch is that this is an environment-specific problem, since we were both able to watch it work on Windows this morning, and the output is from my machine (running macOS) this evening.

Reply:
I actually just figured out the problem! `$GOPATH/bin` wasn't in my `$PATH`, adding it fixed the issue